packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -152,8 +152,28 @@ mixin CupertinoRouteTransitionMixin<T> on PageRoute<T> {
 
   @override
   bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
+    if (controller != null && controller!.isAnimating) {
+      return false;
+    }
```

Comment:
Doesn't this make overlapping transitions (when you have routes popping and pushing at the same time) more jaring?